Turn <popup> into <div popup> (#518)
openui/open-ui: research/src/pages/prototypes/selectmenu.mdx

</style>
<selectmenu class="my-custom-select">
  <div slot="listbox">
    <div popup="popup" behavior="listbox">
      <option>Option 1</option>
      <option>Option 2</option>
      <option>Option 3</option>
      <option>Option 4</option>
      <option>Option 5</option>
    </div>
  </div>
</selectmenu>
```

> The `<div popup>` used here is the element proposed at [Popup (Editor's Draft)](http://localhost:8000/components/popup). The element with `behavior="listbox"` is required to be a `<div popup>`. Applying `behavior="listbox"` tells the browser to apply select menu listbox behavior to that element: it will be opened when the `<selectmenu>`'s button is clicked, and the user can select `<option>`s inside it with mouse, arrow keys, and touch.

Reviewer: should this be `<div popup=popup>` to match the examples used in the popup explainer?

<img width="404" alt="image" src="https://user-images.githubusercontent.com/17087167/163999249-48b80ac0-1a1b-420b-a525-2e70a4a29e82.png">

Author: Thank you for catching this! I had missed that spot.

> The `<div popup=popup>` used here is the element proposed at [Popup (Editor's Draft)](http://localhost:8000/components/popup). The element with `behavior="listbox"` is required to be a `<div popup=popup>`. Applying `behavior="listbox"` tells the browser to apply select menu listbox behavior to that element: it will be opened when the `<selectmenu>`'s button is clicked, and the user can select `<option>`s inside it with mouse, arrow keys, and touch.

Reviewer 2: I think the text is ok as-is, but it raises a good question: are we going to require that the element with `behavior=listbox` is **exactly** a `<div popup=popup...` or maybe any element that supports `popup=popup`?

@dandclark WDYT? Should we open an issue for this question?

Reviewer 3: Hmm, if we have a dedicated `<listbox>` element it might make sense to limit it to only `<listbox>`.
Otherwise, I think it should probably be `<div>` or `<span>` only. I can't think of any semantically-meaningful elements that would fit.

> The `<div popup=popup>` used here is the element proposed at [Popup (Editor's Draft)](http://localhost:8000/components/popup). The element with `behavior="listbox"` is required to be a `<div popup=popup>`. Applying `behavior="listbox"` tells the browser to apply select menu listbox behavior to that element: it will be opened when the `<selectmenu>`'s button is clicked, and the user can select `<option>`s inside it with mouse, arrow keys, and touch.

Reviewer 3: ```suggestion
The `popup` attribute used in this example's `<div popup=popup>` is from the proposal at [Popup API (Explainer)](https://open-ui.org/components/popup.research.explainer). The element with `behavior="listbox"` is required to be a `<div popup=popup>`. Applying `behavior="listbox"` tells the browser to apply select menu listbox behavior to that element: it will be opened when the `<selectmenu>`'s button is clicked, and the user can select `<option>`s inside it with mouse, arrow keys, and touch.
```
